compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/checkers/AggregationChecker.kt

```diff
@@ -169,10 +174,32 @@ internal object AggregationChecker : FirClassChecker(MppCheckerKind.Common) {
             }
           }
         }
+          if (!checkVisibility(declaration, annotation, session, context, reporter)) return
       }
     }
   }
 
+    private fun checkVisibility(
+        declaration: FirClass,
+        annotation: FirAnnotation,
+        session: FirSession,
+        context: CheckerContext,
+        reporter: DiagnosticReporter
+    ): Boolean {
+        val scopeVisibility =
+            annotation.getKClassArgument(name = "scope".asName(), session)?.toClassSymbol(session)?.visibility
+        val declarationVisibility = declaration.visibility
+        val visibilityComparison = scopeVisibility?.let { declarationVisibility.compareTo(it) }
```

Comment:
Just double checking - are you sure these are (intentionally) ordered by visibility?

Reply:
From `org.jetbrains.kotlin.descriptors.Visibilities`, this is the value map used to make the comparison:
<img width="551" alt="Screenshot 2025-04-09 at 8 07 31 PM" src="https://github.com/user-attachments/assets/4b1dff4b-fff4-4f3d-809a-1e08d11a19e5" />

Reply:
But you're not using that class here correcty? Shouldn't this be

```suggestion
        val visibilityComparison = scopeVisibility?.let { Visibilities.compare(declarationVisibility, it) }
```

Reply:
Not using it directly, but `Visibility.compareTo()` calls `Visibilities.compareLocal()` which uses the map in the screenshot to make the comparison.
<img width="447" alt="Screenshot 2025-04-09 at 9 09 42 PM" src="https://github.com/user-attachments/assets/b2ec6a04-fca2-4ebd-833b-7d4593159648" />
<img width="721" alt="Screenshot 2025-04-09 at 9 09 51 PM" src="https://github.com/user-attachments/assets/3b427e64-f5ce-45c0-803c-a36d1282a0a0" />

Reply:
Oh weird, I just noticed `Visibilities.compare()`, which calls `Visibility.compareTo()` (which calls `Visibilities.compareLocal()`) and then reverses the comparison on a `null`... So I guess maybe that's better? I'm not really sure when `Visibilities.compare()` would actually use `oppositeResult`, though.
<img width="495" alt="Screenshot 2025-04-09 at 9 17 17 PM" src="https://github.com/user-attachments/assets/ceae2d28-1fcb-43a3-b67b-c972af524f8a" />

Reply:
I'm fine with either as long as it's correct!